pyproject.toml

```diff
@@ -1,3 +1,99 @@
+[tool.poetry]
+name = "baseplate"
+version = "2.7.0a1"
+description = "reddit's python service framework"
+authors = ["reddit"]
+license = "BSD"
+readme = "README.md"
+homepage = "https://github.com/reddit/baseplate.py"
+repository = "https://github.com/reddit/baseplate.py"
+documentation = "https://baseplate.readthedocs.io/en/stable/"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: POSIX :: Linux",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Software Development :: Libraries :: Application Frameworks",
+]
+
+[tool.poetry.dependencies]
+python = ">=3.8.1"
+advocate = { version = ">=1.0.0,<2.0", optional = true }
+boto3 = ">=1.28.27"
+cassandra-driver = { version = ">=3.29.0,<4.0", optional = true }
+confluent-kafka = { version = ">=2.3.0", optional = true }
+gevent = ">=23.9.1"
+kazoo = { version = ">=2.5.0,<3.0", optional = true }
+kombu = { version = ">=5.3.3", optional = true }
+objgraph = { version = ">=3.6.0", optional = true }
+posix-ipc = ">=1.0.0,<2.0"
+prometheus-client = ">=0.12.0"
+pymemcache = { version = ">=1.3.0,<1.4.4", optional = true }
+pyramid = { version = ">=1.10.8,<2.0", optional = true }
+python-json-logger = ">=2.0.5,<3.0"
+reddit-cqlmapper = { version = ">=0.3.0,<1.0", optional = true }
+redis = { version = ">=2.10.0,<4.0.0", optional = true }
+redis-py-cluster = { version = ">=2.1.2,<3.0.0", optional = true }
+requests = ">=2.21.0,<3.0"
+sentry-sdk = { version = ">=1.35.0,<2.0", optional = true }
+sqlalchemy = { version = ">=1.4.49,<2", optional = true }
+thrift-unofficial = ">=0.19.0,<1.0"
+
+[tool.poetry.extras]
+amqp = ["kombu"]
+cassandra = ["cassandra-driver"]
+cqlmapper = ["reddit-cqlmapper"]
+kafka = ["confluent-kafka"]
+memcache = ["pymemcache"]
+prometheus = []  # Kept for backwards compatibility, these are now main requirements
+pyramid = ["pyramid"]
+redis = ["redis"]
+redis-py-cluster = ["redis-py-cluster"]
+refcycle = ["objgraph"]
+requests = ["advocate"]
+s3fetcher = []  # Kept for backwards compatibility, these are now main requirements
```

Comment:
is this needed given minor release?

Reply:
I'll keep this for now since I'm trying to keep the generated wheels 1:1 identical with this PR but we can probably drop this. My impression has been that most people aren't using the extras anyway.